lib/puppet/util/windows/adsi.rb

```diff
@@ -94,6 +94,23 @@ def execquery(query)
       wmi_connection.execquery(query)
     end
 
+    # https://docs.microsoft.com/en-us/dotnet/api/microsoft.powershell.commands.domainrole?view=powershellsdk-1.1.0
+    DOMAIN_ROLES = {
+      STANDALONE_WORKSTATION: 0,
+      MEMBER_WORKSTATION: 1,
+      STANDALONE_SERVER: 2,
+      MEMBER_SERVER: 3,
+      BACKUP_DOMAIN_CONTROLLER: 4,
+      PRIMARY_DOMAIN_CONTROLLER: 5,
+    }
+
+    def domain_role
+      query = Puppet::Util::Windows::ADSI.execquery('select DomainRole from Win32_ComputerSystem').to_enum.first
```

Comment:
can `execquery('select DomainRole from Win32_ComputerSystem')` return nil?

Reply:
Looks like it always returns a WIN32OLE object:

```
irb(main):007:0> Puppet::Util::Windows::ADSI.execquery('select DomainRole from Win32_ComputerSystem')
=> #<WIN32OLE:0x00000000059e7618>
irb(main):008:0> Puppet::Util::Windows::ADSI.execquery('select DomainRole from Win32_ComputerSys')
=> #<WIN32OLE:0x00000000059d4e50>
irb(main):009:0> Puppet::Util::Windows::ADSI.execquery('select DomainRole from Win32_ComputerS')
=> #<WIN32OLE:0x00000000041b9af0>
```